libs/wingsdk/src/target-awscdk/bucket.ts

```diff
@@ -0,0 +1,87 @@
+import { Construct } from "constructs";
+import * as cloud from "../cloud";
+import * as core from "../core";
+import { BlockPublicAccess, BucketEncryption, Bucket as S3Bucket } from "aws-cdk-lib/aws-s3";
+import { BucketDeployment, Source } from "aws-cdk-lib/aws-s3-deployment";
+import { CaseConventions, NameOptions, ResourceNames } from "../utils/resource-names";
+import { RemovalPolicy } from "aws-cdk-lib";
+
+export const BUCKET_PREFIX_OPTS: NameOptions = {
+  maxLen: 37,
+  case: CaseConventions.LOWERCASE,
+  disallowedRegex: /([^a-z0-9\-]+)/g,
+  // add a dash to the end of the prefix to distinguish between the
+  // Wing-generated portion of the name and the suffix generated by Terraform
+  // suffix: "-",
+};
+
+/**
+ * AWS implementation of `cloud.Bucket`.
+ *
+ * @inflight `@winglang/sdk.cloud.IBucketClient`
+ */
+export class Bucket extends cloud.BucketBase {
+  private readonly bucket: S3Bucket;
+  private readonly public: boolean;
+
+  constructor(scope: Construct, id: string, props: cloud.BucketProps) {
+    super(scope, id, props);
+
+    this.public = props.public ?? false;
+
+    const bucketName = ResourceNames.generateName(this, BUCKET_PREFIX_OPTS);
```

Comment:
Contrary to terraform, the recommended practice for cloudformation (cdk) is not to supply a name at all and let the engine allocate one.

So this is not needed

Reply:
I can remove it, but I'm a little worried about this change. Personally, I never let the CDK create my resource names because they are difficult to find in the console.